src/Microsoft.Health.Dicom.Core/Features/Query/QueryResponseBuilder.cs

```diff
@@ -106,13 +106,28 @@ public QueryResponseBuilder(QueryExpression queryExpression)
     }
 
     public DicomDataset GenerateResponseDataset(DicomDataset dicomDataset)
+    {
+        return GenerateResponseDataset(dicomDataset, null);
+    }
+
+    public DicomDataset GenerateResponseDataset(DicomDataset dicomDataset, DicomDataset mergeDataset)
     {
         EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
 
         dicomDataset.Remove(di => !_tagsToReturn.Any(
             t => t.Group == di.Tag.Group &&
             t.Element == di.Tag.Element));
 
+        if (mergeDataset != null)
```

Comment:
Shouldn't we return the filtercondition fields too? 

For e.g if the query is https://localhost:63838/v1/studies?ModalitiesInStudy=PT. Don't we need to include ModalitiesInStudy too in return dataset?

Reply:
Yes, it is included. This code is just merging the results from blob and sql. _tagsToReturn should already have filter field from existing logic